pkg/transport/generic_transport_controller.go

```diff
@@ -0,0 +1,512 @@
+		UpdateFunc: func(_, new interface{}) { transportController.handleWrappedObject(new) },
+		DeleteFunc: transportController.handleWrappedObject,
+	})
+	dynamicInformer.Start(ctx.Done())
+
+	return transportController, nil
+}
+
+func convertObjectToUnstructured(object runtime.Object) (*unstructured.Unstructured, error) {
+	unstructuredObject, err := runtime.DefaultUnstructuredConverter.ToUnstructured(object)
+	if err != nil {
+		return nil, fmt.Errorf("failed to convert given object to unstructured - %w", err)
+	}
+	return &unstructured.Unstructured{Object: unstructuredObject}, nil
+}
+
+func getGvrFromWrappedObject(restConfig *rest.Config, wrappedObject runtime.Object) (*schema.GroupVersionResource, error) {
+	unstructuredWrappedObject, err := convertObjectToUnstructured(wrappedObject)
+	if err != nil {
+		return nil, fmt.Errorf("failed to convert wrapped object to unstructured - %w", err)
+	}
+
+	gvk := unstructuredWrappedObject.GroupVersionKind()
+	clientset, err := kubernetes.NewForConfig(restConfig)
+	if err != nil {
+		return nil, fmt.Errorf("failed to create k8s clientset for given config - %w", err)
+	}
+	mapper := restmapper.NewDeferredDiscoveryRESTMapper(cacheddiscovery.NewMemCacheClient(clientset.Discovery()))
+
+	restMapping, err := mapper.RESTMapping(gvk.GroupKind(), gvk.Version)
+	if err != nil {
+		return nil, fmt.Errorf("failed to discover GroupVersionResource from given GroupVersionKind - %w", err)
+	}
+
+	return &(restMapping.Resource), nil
+}
+
+type genericTransportController struct {
+	logger logr.Logger
+
+	placementDecisionLister         edgev1alpha1listers.PlacementDecisionLister
+	placementDecisionInformerSynced cache.InformerSynced
+	wrappedObjectInformerSynced     cache.InformerSynced
+
+	// workqueue is a rate limited work queue.
+	// This is used to queue work to be processed instead of performing it as soon as a change happens.
+	// This means we can ensure we only process a fixed amount of resources at a time, and makes it
+	// easy to ensure we are never processing the same item simultaneously in two different workers.
+	workqueue workqueue.RateLimitingInterface
+
+	//transport is a specific implementation for the transport interface.
+	transport        Transport
+	transportClient  dynamic.Interface // dynamic client to transport wrapped object. since object kind is unknown during complilation, we use dynamic
+	wrappedObjectGVR schema.GroupVersionResource
+
+	wdsClientset     *edgeclientset.Clientset
+	wdsDynamicClient *dynamic.DynamicClient
+	wdsName          string
+}
+
+// enqueuePlacementDecision takes an PlacementDecision resource and
+// converts it into a namespace/name string which is put onto the workqueue.
+// This func *shouldn't* handle any resource other than PlacementDecision.
+func (c *genericTransportController) enqueuePlacementDecision(obj interface{}) {
+	var key string
+	var err error
+	if key, err = cache.MetaNamespaceKeyFunc(obj); err != nil {
```

Comment:
If this line were to call https://github.com/kubernetes/client-go/blob/v0.28.2/tools/cache/store.go#L121 instead and the queue held `ObjectName` values then ` func (c *genericTransportController) process` could be simpler --- no parsing needed there.

Reply:
I prefer using the `cache.DeletionHandlingMetaNamespaceKeyFunc` function which covers also the case of `cache.DeletedFinalStateUnknown` and have only this single handling function for all event types.